lib/custom/src/MShop/Service/Provider/Payment/Payone.php

```diff
@@ -22,6 +22,52 @@ class Payone
 	implements \Aimeos\MShop\Service\Provider\Payment\Iface
 {
 	/**
+	 * Returns the data passed to the Omnipay library
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Base\Iface $base Basket object
+	 * @param $orderid Unique order ID
+	 * @param array $params Request parameter if available
+	 */
+	protected function getData( \Aimeos\MShop\Order\Item\Base\Iface $base, $orderid, array $params )
+	{
+		$data = parent::getData( $base, $orderid, $params );
+		$lines = [];
+		foreach( $base->getProducts() as $product )  {
+			$lines[] = new \Omnipay\Payone\Extend\Item([
+				'id' => (string) $product->toArray()['order.base.product.prodcode'],
+				'name' => $product->getName(),
+				'itemType' => 'goods', // Available types: goods, shipping etc.
+				'quantity' => $product->getQuantity(),
+				'price' => $product->getPrice()->getValue(),
+				'vat' => (int) $product->getPrice()->getTaxRate(),
+			]);
+		}
+		if ($base->getService('delivery')->getPrice()->getCosts() != '0.00') {
+			$deliveryObject = $base->getService('delivery');
+			$lines[] = new \Omnipay\Payone\Extend\Item([
+				'id' => (string) $deliveryObject->getId(),
+				'name'->getService('delivery')->getName()
```

Comment:
There's a comma missing so the unit tests fail: https://travis-ci.org/aimeoscom/ai-payments/builds/236277601